### ISSUE TITLE
Improve "Add Equation" dialog

### ISSUE THREAD

**Issue by bertob** [SjoerdB93/Graphs#108]
The layout of the "add equation" dialog could be better, a list of grouped AdwEntries isn't ideal here, especially when some of them don't need to be quite so large.

![image](https://user-images.githubusercontent.com/1908896/217405468-9de02fe9-afa8-45d7-b6c3-b724dee14e69.png)

Here's a quick sketch for what a more appropriate layout could look like:

![image](https://user-images.githubusercontent.com/1908896/217405633-94346070-c4e4-4182-a59c-d5393ee83691.png)

**Comment by SjoerdB93** [comment]
Thanks a lot for the feedback. 
I made a quick variant with this feedback in mind, and I do have to say it's already quite an improvement. See screenshot:
![image](https://user-images.githubusercontent.com/68477016/217449625-dff76b07-a679-405d-883f-ddbb2487484d.png)

I've also tried to set the `halign` to `center` as in the sketch, but it looked quite off. Maybe decreasing the margins there will improve ti but I was a bit unsure.
![image](https://user-images.githubusercontent.com/68477016/217449755-1b6f657f-57bd-46b5-9916-26c07e8cb89c.png)

I may also want to decrease the default size, then it looks like this (with the centered button):
![image](https://user-images.githubusercontent.com/68477016/217449873-12cd291c-ea87-4f43-99dc-62ea6d2ab8f8.png)

I've uploaded this as a separate branch and a draft PR.

**Comment by bertob** [comment]
Looks pretty good, a few notes:
- I'd drop the second "Add Equation" title above the formula
- It'd look better with a flat header bar, like e.g. AdwAboutWindow
- The button at the bottom should have the `.pill` class and maybe have a slightly wider minimum width, but I wouldn't go full-width
- I'd double the spacing between the name and the 3 entries below
- The spacing between the 3 entries at the bottom should be the same in both gaps (12px)

**Comment by SjoerdB93** [comment]
Thanks, I'll take a look and try apply the points in this list later today.  Also agree about the flat header bar

On a similar note, I'll probably change the "Transform Data" dialog and the "Add data (advanced)" dialog a bit as well to match the updated design language in "Add equation" a bit more.

**Comment by SjoerdB93** [comment]
That's already an improvement

![image](https://user-images.githubusercontent.com/68477016/217625137-ce99cc69-9d8a-433c-9e31-f8e29f1f8b12.png)

I've given the "Transform Data" dialog and "Add Data (advanced)" dialog a flat header bar as well to match this style.

**Comment by bertob** [comment]
This looks good, I'd just slightly weak the vertical spacing above the pill button to 1.5-2x the current value, and make the window a little bit taller so the margin at the bottm matches the margins on the sides.

**Comment by SjoerdB93** [comment]
Thanks, this current version has a margin of 26 px on all sides.

![image](https://user-images.githubusercontent.com/68477016/217735746-fe62f908-b66b-4b0d-b6d9-e55fe95d48d4.png)

**Comment by SjoerdB93** [comment]
I've updated the other modal dialogs now as well in PR #111, I'm a bit unsure about Transform Window though. I've stashed the extra information in an AdwExpanderRow, but can't get the window to resize back to the original size when the user undoes the expansion (I can get it to execute a function when toggled, but I can't get the windows to resize programmatically).

Not sure if it's the worst thing in the world, but it feels a bit unpolished with the window keeping the larger size maybe.

**Comment by bertob** [comment]
The equation dialog looks good to me, except for the missing spacing between the first entry and the label below (I wonder what's broken there, on your screenshot it looks fine), and this issue where the horizonally arranged entries resize depending on which has focus:

![image](https://user-images.githubusercontent.com/1908896/217819211-6e499b3d-7a0a-4536-b7a3-3e1cd5a1e119.png)

The layout on the advanced dialog could be better (I'd maybe split the rows up into sections rather than one big list), but that's maybe for another MR.

Also, there are some non-Title Case labels, e.g. "Add files" instead of "Add Files". Please make sure row titles and button labels laways use title case.

**Comment by SjoerdB93** [comment]
The missing margin was because the screenshot was from a slightly older build than the latest commit, but I reverted that change and out that margin back in.

Regarding the advanced add files dialog. It think you're right about splitting this into sections would be an improvement. I'll see if I can find some time tomorrow to try and tackle that.

Not sure about the issue of selected entry rows in the bottom row becoming smaller. I tried several things but couldn't fix it. I'll take a look later or else I'll try and ask around. It's not the worst bug in the world, but would be nice to keep that static i think.
